### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by vietcrit96**
screenshot from 2019-01-03 14-40-47

### CODE REVIEW

**Comment by vietcrit96** [comment]
ready

File: app/assets/stylesheets/sessions.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (thanhhungchu95) [inline app/assets/stylesheets/sessions.scss]: remove this file

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -3,15 +3,36 @@ def log_in user
...
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by id: user_id
+    elsif user_id = cookies.signed[:user_id]
+      user = User.find_by id: user_id
+      if user&.authenticated?(cookies[:remember_token])
```
Comment (thanhhungchu95) [inline app/helpers/sessions_helper.rb:18]: add an empty line above

File: app/models/user.rb [hunk 3]
```diff
@@ -9,13 +10,32 @@ class User < ApplicationRecord
     length: {minimum: Settings.user_pass_length_min}
   before_save{email.downcase!}
   has_secure_password
+  class << self
```
Comment (thanhhungchu95) [inline app/models/user.rb:14]: add an empty line above

File: app/models/user.rb [hunk 4]
```diff
@@ -9,13 +10,32 @@ class User < ApplicationRecord
...
+    end
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
```
Comment (thanhhungchu95) [inline app/models/user.rb]: use `update` method

File: app/models/user.rb [hunk 5]
```diff
@@ -9,13 +10,32 @@ class User < ApplicationRecord
...
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
```
Comment (thanhhungchu95) [inline app/models/user.rb]: use update method

File: app/views/sessions/new.html.erb [hunk 6]
```diff
@@ -6,10 +6,15 @@
     <%= form_with scope: :session, url: login_path, local: true do |f| %>
 
       <%= f.label :email %>
-      <%= f.email_field :email, class: "form-control" %>
+      <%= f.email_field :email, class: 'form-control' %>
```
Comment (thanhhungchu95) [inline app/views/sessions/new.html.erb]: double-quote

**Comment by vietcrit96** [comment]
ready

**Review by thanhhungchu95 — COMMENTED** [review]
(merged) đi chế =))

**Pull request merged**